code/WorkInProgress/zamu_mail.dm

```diff
@@ -269,23 +326,100 @@
 
 		// make a gift for this person
 		var/obj/item/random_mail/package = null
-		var/package_color = "#FFFFFF"
+		var/package_color = recipient["fav_color"] ? recipient["fav_color"] : pick("#FFFFAA", "#FFBB88", "#FF8800", "#CCCCFF", "#FEFEFE")
+		var/stamp_id = "stamp-stain-[rand(1, 3)]"
+		var/sender = null
+		var/sender_title = null
+		var/list/context = list()
+
+		// high chance to find an appropriate mail based on the traits chosen
+		var/mob/living/carbon/human/H = picked
+		if (ishuman(picked))
+			H = picked
+		if (prob(50) && length(H?.traitHolder.traits) > 0)
```

Comment:
I'm not sure I'm huge on this being a flat 50% chance to replace the job-specific table while having exactly one trait. I think a percentmult-based prob scaled by traitcount might be a better approach (though still has some issues e.g.: "training_foo" traits counting towards it but meh)
e.g.:
```suggestion
		if (prob(percentmult(20, length(H?.traitHolder.traits)) * 0.6))
```